libc/utils/docgen/docgen.py

```diff
@@ -23,12 +23,19 @@ def load_api(hname: str) -> Dict:
 # TODO: we may need to get more sophisticated for less generic implementations.
 # Does libc/src/{hname minus .h suffix}/{fname}.cpp exist?
 def is_implemented(hname: str, fname: str) -> bool:
-    return Path(
+    path = Path(
         Path(__file__).parent.parent.parent,
         "src",
-        hname.rstrip(".h"),
-        fname + ".cpp",
-    ).exists()
+        hname.rstrip(".h")
+    )
+    source_file_name = fname + ".cpp"
+    # Recursively search for the target source file in the subdirectories under
+    # libc/src/{hname}.
+    for sub_dir in path.glob("**"):
+        if sub_dir.joinpath(source_file_name).exists():
+            return True
+
+    return False
```

Comment:
Is it possible to `glob(f"**/{source_file_name}"` and then see if that's empty or not?